File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/ObjectType.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum ObjectType {
    TRASH_BIN("Çöp Kutusu"),
    BENCH("Bank"),
    TREE("Ağaç"),
    LIGHTING_POLE("Aydınlatma Direği"),
    PLAYGROUND_EQUIPMENT("Oyun Alanı Ekipmanı");

    private final String displayName;

    ObjectType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: frontend/lib/models/enum/object_type.dart
enum ObjectType {
  trashBin('Çöp Kutusu', 'TRASH_BIN'),
  bench('Bank', 'BENCH'),
  tree('Ağaç', 'TREE'),
  lightingPole('Aydınlatma Direği', 'LIGHTING_POLE'),
  playgroundEquipment('Oyun Alanı Ekipmanı', 'PLAYGROUND_EQUIPMENT');

  final String displayName;
  final String apiValue;

  const ObjectType(this.displayName, this.apiValue);

  static ObjectType fromApiValue(String value) {
    return ObjectType.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen Cisim Türü: $value'),
    );
  }
}
